bndtools.builder/src/org/bndtools/builder/ComponentMarker.java

```diff
@@ -0,0 +1,249 @@
+package org.bndtools.builder;
+
+import java.io.File;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.bndtools.api.BndtoolsConstants;
+import org.bndtools.api.ILogger;
+import org.bndtools.api.Logger;
+import org.bndtools.builder.decorator.ui.ComponentDecorator;
+import org.bndtools.builder.decorator.ui.ComponentPackageDecorator;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IImportDeclaration;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMemberValuePair;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.Document;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IDecoratorManager;
+import org.eclipse.ui.PlatformUI;
+
+import aQute.bnd.build.Project;
+
+/**
+ * This class creates markers for classes that contain the {@link org.osgi.service.component.annotations.Component}
+ * annotation, and stores this information in the {@link BuilderPlugin} for use by {@link ComponentDecorator}and
+ * {@link ComponentPackageDecorator}.
+ *
+ * @author wodencafe
+ */
+public class ComponentMarker {
+
+    private static final ILogger logger = Logger.getLogger(ComponentMarker.class);
+    public static final String ANNOTATION_COMPONENT_PACKAGE = "org.osgi.service.component.annotations";
+    public static final String ANNOTATION_COMPONENT_FQN = ANNOTATION_COMPONENT_PACKAGE + ".Component";
+
+    private final static ConcurrentMap<String,ComponentDecorationToken> compMap = new ConcurrentHashMap<>();
```

Comment:
Hi @gamerson, I can look into this change, I have to do some more reading on IMarkers :)